pkg/controllers/user/logging/utils/templatewrap.go

```diff
@@ -304,3 +342,81 @@ func parseEndpoint(endpoint string) (host string, scheme string, err error) {
 
 	return u.Host, u.Scheme, nil
 }
+
+func ValidateCustomTargetContent(content string, enableEscaped bool) (string, error) {
+	r := strings.NewReader(content)
+	buf := bufio.NewReader(r)
+	var newContent bytes.Buffer
+	for {
+		l, err := buf.ReadString('\n')
+		if l != "" {
+			line := strings.TrimSpace(l)
+			lineArray := strings.Split(line, " ")
+			if len(lineArray) == 0 {
+				continue
+			}
+
+			key := lineArray[0]
+			if err := filterFluentdTags(key); err != nil {
+				return "", err
+			}
+
+			if enableEscaped {
+				value := strings.TrimSpace(strings.TrimPrefix(line, key))
+				newValue := escapeString(value)
+				newContent.WriteString(key)
+				newContent.WriteString(" ")
+				newContent.WriteString(newValue)
+				newContent.WriteString("\n\t")
+			}
+		}
+
+		if err == io.EOF {
+			return newContent.String(), nil
+		}
+	}
+}
+
+func ValidateCustomTags(tags map[string]string, enableEscaped bool) (map[string]string, error) {
+	newCustomTags := make(map[string]string)
+	for key, value := range tags {
+		if err := filterFluentdTags(key); err != nil {
+			return nil, err
+		}
+
+		if enableEscaped {
+			newValue := escapeString(value)
+			newCustomTags[key] = newValue
+		}
+	}
+
+	return newCustomTags, nil
+}
+
+func filterFluentdTags(key string) error {
+	invalidTagKeys := []string{
+		"<source", "<parse", "<filter", "<format", "<storage", "<buffer", "<match",
+		"</source>", "</parse>", "</filter>", "</format>", "</storage>", "</buffer>", "</match>",
+	}
+
+	for _, invalidKey := range invalidTagKeys {
+		if strings.Contains(key, invalidKey) {
+			return errors.New("invalid custom tag key: " + key)
+		}
+	}
+
+	return nil
+}
+
+func escapeString(s string) string {
+	var escapeReplacer = strings.NewReplacer(
+		"\t", `\\t`,
+		"\n", `\\n`,
+		"\r", `\\r`,
+		"\"", `\\\"`,
+		"\\", `\\\\`,
```

Comment:
Should it be `\` -> `\\`

Reply:
may also cause miss convert, like ```\b``` -> ```\\b```, ```\123``` ->```\\123```

Reply:
No, the [fluentd document](https://docs.fluentd.org/v0.12/articles/config-file) says that it should have.
> `\` is interpreted as escape character. You need `\` for setting `"`, `\r`, `\n`, `\t`, `\` or several characters in double-quoted string literal.